### ISSUE TITLE
Save taxonomy fields data to custom db table

### ISSUE THREAD

**Issue by klocus** [TypeRocket/typerocket#176]
Hello,

My plugin is creating a dynamic taxonomies which depend on user input, so I can't create a Controller and Model for each taxonomy. It makes me unable to add additional fields to created taxonomies, because taxonomy need a Controller and Model for this.

I've already tried something like this:
```php
foreach ($this->fields as $key => $value) {
    if ($value['active'] && $value['type'] == 'taxonomy') {
        tr_taxonomy($key)
        ->addPostType('cpt')
        ->setMainForm(function() {
            $form = tr_form('my_custom_controller');
            echo $form->text('test');
        });
    }
} 
```

But it doesn't work. How can I achieve my goal? How to use only one Term Controller (or other) for all Taxonomies?

**Comment by kevindees** [comment]
Hey @klocus 

You may be able to do this using the `tr_model` action hook but I'm not sure.

Thanks,
Kevin

**Comment by klocus** [comment]
Hi @kevindees,

Thanks for your advice, but it is still not possible :/ Model is changing but not controller.

```php
foreach ($this->fields as $key => $value) {
    tr_taxonomy($key)
    ->addPostType('cpt')
    ->setMainForm(function() {
        if (isset($GLOBALS['taxonomy']) && isset($GLOBALS['tag_ID'])) {
            $form = tr_form('my_custom_controller', 'update', $GLOBALS['tag_ID']); 
        } else {
            $form = tr_form('my_custom_controller', 'create');
        }
        echo $form->text('test');
    });

    // Change model's taxonomy
    add_action('tr_model', function($model) {
        $model->taxonomy = $GLOBALS['taxonomy'];
    });
}
```

`tr_form` doesn't change the controller. It still use taxonomy name.

I added a new method to `TypeRocket\Register\Taxonomy`:

```php
public function setResource( array $resource )
{
    $this->resource = $resource;

    return $this;
}
```

But maybe there is a way without adding a new method?

**Comment by nebulousGirl** [comment]
Maybe you could create the files from the user input.

**Comment by kevindees** [comment]
Hey @klocus 

I recommend adding a new base Controller class with the methods you need.

```php
<?php
namespace App\Controllers;

use \App\Models\Category;
use \App\Controllers\UserDefinedTermController;

class CategoryController extends UserDefinedTermController
{
    protected $modelClass = Category::class;
}
```

```php
<?php
namespace App\Controllers;

class UserDefinedTermController extends WPTermController
{
    protected function init()
    {
        parent::init();
        // do stuff;
    }

    public function setResource( array $resource )
    {
        $this->resource = $resource;

        return $this;
    }
}
```

Or, maybe simply this would work for you.

```php
<?php
namespace App\Controllers;

use \App\Models\Category;
use \TypeRocket\Controllers\WPTermController;

class CategoryController extends WPTermController
{
    protected $modelClass = Category::class;

    protected function init() {
        parent::init();
        // do stuff;
    }
}
```

**Comment by klocus** [comment]
@kevindees in my opinion, it is impossible to change the resource of `class Taxonomy extends Registrable` on that way. Class `Taxonomy` is not connected to `WPTermController`, so it cannot change the resource.

**Comment by kevindees** [comment]
@klocus 

I have made a number of upgrades and fixes to the WPTerm class in v4.0.4. Let me know if that helps you.

https://github.com/TypeRocket/core/releases/tag/v4.0.4
